### PR TITLE
tests: add go.mod file to isolate

### PR DESCRIPTION
**Pull request by jmank88**
This PR proposes isolating `tests/` as a separate module, like we did in the core repo. This removes more than 100 dependencies from the main `go.mod`!
[Terra version](https://github.com/smartcontractkit/chainlink-terra/pull/281)

Does CI need updating?

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,134 +2,72 @@ module github.com/smartcontractkit/chainlink-solana
...
-	github.com/smartcontractkit/helmenv v1.0.68
-	github.com/smartcontractkit/libocr v0.0.0-20220414173908-cdfa6bef133a
-	github.com/stretchr/testify v1.7.1
-)
-
-require github.com/satori/go.uuid v1.2.0
-
 require (
 	github.com/davecgh/go-spew v1.1.1
-	github.com/ethereum/go-ethereum v1.10.16
```
Comment (jmank88) [inline go.mod:18]: ❌

**Review by aalu1418 — DISMISSED** [review]
This is amazing 🎉

**Review by krebernisak — CHANGES_REQUESTED** [review]
Please add a README to the `./tests` folder explaining the split.

**Comment by jmank88** [comment]
Added readme and created #335 to follow up.

**Review by aalu1418 — APPROVED** [review]
🔥

**Review by krebernisak — APPROVED** [review]
👏

**Pull request merged**